// @ediri/vultr/reverseIpv4.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReverseIpv4 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Provides a Vultr Reverse IPv4 resource. This can be used to create, read, and
 * modify reverse DNS records for IPv4 addresses. Upon success, DNS
 * changes may take 6-12 hours to become active.
 *
 * ## Example Usage
 *
 * Create a new reverse DNS record for an IPv4 address:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as vultr from "@ediri/vultr";
 *
 * const myInstance = new vultr.Instance("myInstance", {
 *     enableIpv6: true,
 *     osId: 477,
 *     plan: "vc2-1c-2gb",
 *     region: "ewr",
 * });
 * const myReverseIpv4 = new vultr.ReverseIpv4("myReverseIpv4", {
 *     instanceId: myInstance.id,
 *     ip: myInstance.mainIp,
 *     reverse: "host.example.com",
 * });
 * ```
 */
class ReverseIpv4 extends pulumi.CustomResource {
    /**
     * Get an existing ReverseIpv4 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ReverseIpv4(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReverseIpv4.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReverseIpv4.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["gateway"] = state ? state.gateway : undefined;
            resourceInputs["instanceId"] = state ? state.instanceId : undefined;
            resourceInputs["ip"] = state ? state.ip : undefined;
            resourceInputs["netmask"] = state ? state.netmask : undefined;
            resourceInputs["reverse"] = state ? state.reverse : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.instanceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceId'");
            }
            if ((!args || args.ip === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ip'");
            }
            if ((!args || args.reverse === undefined) && !opts.urn) {
                throw new Error("Missing required property 'reverse'");
            }
            resourceInputs["instanceId"] = args ? args.instanceId : undefined;
            resourceInputs["ip"] = args ? args.ip : undefined;
            resourceInputs["reverse"] = args ? args.reverse : undefined;
            resourceInputs["gateway"] = undefined /*out*/;
            resourceInputs["netmask"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ReverseIpv4.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ReverseIpv4 = ReverseIpv4;
/** @internal */
ReverseIpv4.__pulumiType = 'vultr:index/reverseIpv4:ReverseIpv4';
